src/sentry/static/sentry/app/components/events/interfaces/spans/spanBar.tsx

```diff
@@ -370,6 +373,19 @@ class SpanBar extends React.Component<SpanBarProps, SpanBarState> {
             return null;
           }
 
+          const records = Object.values(WEB_VITAL_DETAILS).filter(vital =>
+            names.includes(vital.slug)
+          );
+
+          const failedThreshold =
+            records.find(record => {
+              const value = marks[record.slug];
+              if (typeof value === 'number') {
+                return value >= record.failureThreshold;
+              }
+              return false;
+            }) !== undefined;
```

Comment:
Should we have a a helper function for this logic? It appears twice in these changes.

Reply:
I lifted this up. https://github.com/getsentry/sentry/pull/21910/commits/96de96da95607536130a89f70498a32def61ec62